pkg/controller/remoteingress/remoteingress_controller.go

```diff
@@ -81,22 +120,32 @@ func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
 	return nil
 }
 
+type reconcileContext struct {
+	clusterDeployment *hivev1.ClusterDeployment
+	certBundleSecrets map[string]map[string][]byte
```

Comment:
this seems like an odd struct :) 
in my controller I just gather a slice of secrets and pass that to the function that generates the syncset
I don't think you need to be able to look up the secrets by name since you already get that from the certificateBundles struct
You also lose some info when only obtaining the secret data and not preserving secret type, annotations, labels

Reply:
it took me a while to remember what i was thinking here. i believe what i had in mind was two ingress entries referencing the same certBundle. i was trying to avoid making two secret entries in the syncSet.

i suppose i can still pass the slice with possible duplicate entries, and then when building the syncset, make sure i don't add it twice...

Reply:
This is what I do:
https://github.com/csrwng/hive/blob/master_certs/pkg/controller/controlplanecerts/controlplanecerts_controller.go#L178-L221

Reply:
basically use stringsets to remove dups

Reply:
@csrwng i updated it to just save the slice of secrets, and fixed up the tests accordingly